realm/src/main/java/io/realm/Realm.java

```diff
@@ -1623,10 +1615,6 @@ void setVersion(long version) {
         }
     }
 
-    private <E extends RealmObject> boolean isObjectInRealm(E object) {
```

Comment:
why not using this instead of replicating the code in many places? we could make it package protected instead of private.